### PR TITLE
game/hangman

### PR DESCRIPTION
**Pull request by ArPiiX**

### CODE REVIEW

File: src/app/commands/hangman.py [hunk 1]
```diff
@@ -38,45 +38,57 @@ def main(system: SystemMessenger, author: User, cmd: Command, params: list):
...
+            system.send("Invalid guess! Solve needs to have at least one char!")
             return
-        system.send(no_game())
-        return
 
     if params[0].lower() == "state":
         if hangman_game.get_state():
             system.send(hangman_game.get_word())
-            SHL.output(f"{author} fetched the state of the game.", "HangmanGame")  # log
+            SHL.output(f"{author.display_name} fetched the state of the game.", "HangmanGame")  # log
```
Comment (zaanposni) [inline src/app/commands/hangman.py]: please use `author` only to output to stdout.

File: src/app/commands/hangman.py [hunk 2]
```diff
@@ -38,45 +38,57 @@ def main(system: SystemMessenger, author: User, cmd: Command, params: list):
 
     if params[0].lower() == "start" and params[1].lower() != "":
         if not hangman_game.get_state():
-            hangman_game.reset_game()
-            hangman_game.start(params[1], author)
-            SHL.output(f"{author} started a game with: {params[1]}", "HangmanGame")  # log
-            system.broadcast(f"{author} is challenging everyone to a hangman game!")
-            system.broadcast(f"The word searched is: {hangman_game.get_word()}")
+            if not ("-" in params[1] or "_" in params[1]):
+                hangman_game.reset_game()
```
Comment (zaanposni) [inline src/app/commands/hangman.py]: no need to reset the game, as you do this in your `start()` method

File: src/app/commands/hangman.py (hunk 1 above)
Comment (ArPiiX) [reply src/app/commands/hangman.py]: done

File: src/app/commands/hangman.py (hunk 2 above)
Comment (ArPiiX) [reply src/app/commands/hangman.py]: done

**Pull request merged**